README.md

```diff
@@ -1,44 +1,284 @@
 # Seccompiler
 
-Provides easy-to-use linux [seccomp-bpf](https://www.kernel.org/doc/Documentation/prctl/seccomp_filter.txt)
-jailing for rust-based VMMs.
+Provides easy-to-use linux [seccomp-bpf](https://www.kernel.org/doc/html/latest/userspace-api/seccomp_filter.html) jailing.
+
+Seccomp is a safety net that can be applied on a process, to restrict a
+malicious user, that found a way of running arbitrary code, from executing
+potentially dangerous system calls.
+
+Writing BPF programs by hand is difficult and error-prone. This crate provides
+a library of high-level wrappers over system call filtering, as well as
+functions for generating and installing the filters.
+
+## Supported platforms
+
+Due to the fact that seccomp is a linux-specific feature, this crate is
+supported only on linux systems.
+
+Supported host arhitectures:
+- Little-endian x86_64
+- Little-endian aarch64.
+
+## Short seccomp tutorial
+
+Linux supports seccomp filters as BPF programs, that are interpreted by the
+kernel before each system call.
+
+They are installed in the kernel using 
+[`prctl(PR_SET_SECCOMP)`](https://man7.org/linux/man-pages/man2/prctl.2.html)
```

Comment:
They cannot be invalidated per se. A process can have multiple filters and once a filter is installed on a process, it cannot be uninstalled.
If there are multiple filters for a thread, they are evaluated in the order they were installed and the most restrictive action is taken. i.e. If the first filter kills the process on `exec` but the second filter allows it, the process will be killed.

There are actually some best practices for seccomp (like having an allowlist vs a denylist) but they're mentioned in the man pages for `prctl`/`seccomp` and here https://www.kernel.org/doc/html/latest/userspace-api/seccomp_filter.html. Do you think it's worth having a similar section?

Reply:
my bad, I was assuming that the first filter denied the `exec` while the second allowed it, in which case it cannot be invalidated.
But if it's the other way around, it can be further restricted.
I'll think about adding a `Recommendations` section

Reply:
In the recommendation section you can also link the other recommendations that you found.

Reply:
yes, I added a `Seccomp best practices` section with all recommendations I know of.